.github/workflows/ci.yml

```diff
@@ -1,13 +1,7 @@
 name: CI
 
 on:
-  push:
-    branches:
-      - '*'
-      - '!main'
   pull_request:
```

Comment:
why? I believe we don't want CI to run on every PR because of security issue. Please just use the same condition as we used other places. like https://github.com/awslabs/aws-c-common/blob/main/.github/workflows/ci.yml#L3-L6

Reply:
@TingDaoK this change is part of https://github.com/awslabs/aws-crt-swift/pull/80.